pynest/nest/tests/test_csa.py

```diff
@@ -48,129 +48,125 @@
-        pop1 = nest.LayoutNetwork("iaf_neuron", [n])
+        sources = nest.Create("iaf_psc_alpha", n_neurons)
+        targets = nest.Create("iaf_psc_alpha", n_neurons)
 
         cg = csa.cset(csa.oneToOne)
 
-        nest.CGConnect(pop0, pop1, cg)
+        nest.CGConnect(sources, targets, cg)
 
-        sources = nest.GetLeaves(pop0)[0]
-        targets = nest.GetLeaves(pop1)[0]
-        for i in range(n):
-            conns = nest.GetStatus(
-                nest.GetConnections([sources[i]]), 'target')
+        for i in range(n_neurons):
+            conns = nest.GetStatus(nest.GetConnections([sources[i]]))
             self.assertEqual(len(conns), 1)
-            self.assertEqual(conns[0], targets[i])
+            self.assertEqual(conns[0]["target"], targets[i])
 
-            conns = nest.GetStatus(
-                nest.GetConnections([targets[i]]), 'target')
+            conns = nest.GetStatus(nest.GetConnections([targets[i]]))
             self.assertEqual(len(conns), 0)
 
-    def test_CSA_OneToOne_subnet_nd(self):
-        """One-to-one connectivity with n-dim subnets"""
+    @unittest.skipIf(not HAVE_NUMPY, 'NumPy package is not available')
+    def test_CSA_OneToOne_intvectors(self):
+        """One-to-one connectivity with id intvectors"""
 
         nest.ResetKernel()
 
-        n = 2  # number of neurons per dimension
+        n_neurons = 4
 
-        pop0 = nest.LayoutNetwork("iaf_neuron", [n, n])
-        pop1 = nest.LayoutNetwork("iaf_neuron", [n, n])
+        sources = nest.Create("iaf_psc_alpha", n_neurons)
+        targets = nest.Create("iaf_psc_alpha", n_neurons)
 
         cg = csa.cset(csa.oneToOne)
 
-        self.assertRaisesRegex(nest.NESTError, "BadProperty",
-                               nest.CGConnect, pop0, pop1, cg)
+        nest.CGConnect(numpy.array(sources), numpy.array(targets), cg)
+
+        for i in range(n_neurons):
+            conns = nest.GetStatus(nest.GetConnections([sources[i]]))
+            self.assertEqual(len(conns), 1)
+            self.assertEqual(conns[0]["target"], targets[i])
+
+            conns = nest.GetStatus(nest.GetConnections([targets[i]]))
+            self.assertEqual(len(conns), 0)
 
-    def test_CSA_OneToOne_idrange(self):
-        """One-to-one connectivity with id ranges"""
+    def test_CSA_OneToOne_params(self):
+        """One-to-one connectivity with paramters"""
 
         nest.ResetKernel()
 
-        n = 4  # number of neurons
+        n_neurons = 4
+        weight = 10000.0
+        delay = 2.0
 
-        sources = nest.Create("iaf_neuron", n)
-        targets = nest.Create("iaf_neuron", n)
+        sources = nest.Create("iaf_psc_alpha", n_neurons)
+        targets = nest.Create("iaf_psc_alpha", n_neurons)
 
-        cg = csa.cset(csa.oneToOne)
+        cs = csa.cset(csa.oneToOne, weight, delay)
 
-        nest.CGConnect(sources, targets, cg)
+        nest.CGConnect(sources, targets, cs, {"weight": 0, "delay": 1})
```

Comment:
This could use some comments: You add a cs with a set weight and delay, and here you add these additionally as parameters again. 
In general the tests could use comments to explain why you expect certain things to be like this.

Reply:
Comments added in 84d73c1a3. Thanks for pointing this out. We all get a little blind to these things when working too much with some code, don't we?